template.yaml

```diff
@@ -1,41 +1,59 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Transform: AWS::Serverless-2016-10-31
 Description: >
-  Hello World lambda
-
-Metadata:
-  AWS::ServerlessRepo::Application:
-    Name: "lambda-template"
-    Description: "A GH template for quickly starting a new AWS lambda project."
-    Author: "Sage-Bionetworks"
-    SpdxLicenseId: "Apache-2.0"
-    # paths are relative to .aws-sam/build directory
-    LicenseUrl: "LICENSE"
-    ReadmeUrl: "README.md"
-    Labels: ["serverless", "template", "github", "quick-start"]
-    HomePageUrl: "https://github.com/Sage-Bionetworks-IT/lambda-template"
-    SemanticVersion: "0.0.3"
-    SourceCodeUrl: "https://github.com/Sage-Bionetworks-IT/lambda-template/tree/0.0.3"
+  Lambda to stop or terminate all instances.
 
 # More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
 Globals:
   Function:
     Timeout: 30
 
+Parameters:
+  Schedule:
+    Description: >
+      Schedule to execute the lambda, can be a rate or a cron schedule. Format at
+      https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+    Type: String
+    Default: cron(0 2 * * ? *)  # Run at 2am UTC (8pm PST) every night
+  Ec2Action:
+    Type: String
+    Description: "Action to take on running instances"
+    Default: "STOP"
+    AllowedValues:
+      - "STOP"
+      - "TERMINATE"
```

Comment:
I don't think cloudformation can check for case sensitivity therefore i think  `stop` and `terminate` are also a valid inputs.  If that's true then the python code needs to handle case insensitivity?

Reply:
done

Reply:
am i missing something?  I don't see any updates to handle case insensitivity.

Reply:
The change I made was calling `.lower()` inside `set_instance_state` so that casing is ignored at the time of processing.

Reply:
ohh i see it now.  i thought it would be done in `lambda_handler` right after reading in `EC2_ACTION` so that it's consistent throughout the app.